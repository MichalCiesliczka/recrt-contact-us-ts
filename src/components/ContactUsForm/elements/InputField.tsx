import * as React from 'react';

import TextField from '@material-ui/core/TextField';
import FormHelperText from '@material-ui/core/FormHelperText';
import { WrappedFieldProps } from 'redux-form';

type Props = {
  label: string,
  multiline?: boolean,
} & WrappedFieldProps;

const InputField = ({
  input,
  label,
  multiline,
  meta: {
    touched,
    error,
  },
  ...custom
}: Props) => (
  <div className="input-wrapper">
    <TextField
      fullWidth={true}
      label={label}
      multiline={multiline}
      error={touched && !!error}
      {...input}
      {...custom}
    />
    {
      touched && error && (
        <FormHelperText error={true}>
          {error}
        </FormHelperText>
      )
    }
  </div>
);

export default InputField;
