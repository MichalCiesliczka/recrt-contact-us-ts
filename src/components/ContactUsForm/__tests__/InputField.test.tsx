import * as React from 'react';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';

import InputField from '../elements/InputField';

describe('<InputField />', () => {
  const defaultProps = {
    input: {},
    label: 'Test label',
    meta: {
      error: false,
      touched: false,
    },
    multiline: false,
  };

  it('renders without crashing', () => {
    shallow(<InputField {...defaultProps} />);
  });

  it('should render default status', () => {
    const tree = renderer
      .create(
        <InputField {...defaultProps} />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render error status', () => {
    const preparedProps = {
      ...defaultProps,
      meta: {
        error: true,
        touched: true,
      },
    };

    const tree = renderer
      .create(
        <InputField {...preparedProps} />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
