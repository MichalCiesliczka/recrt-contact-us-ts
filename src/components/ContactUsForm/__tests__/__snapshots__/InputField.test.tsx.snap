// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> should render default status 1`] = `
<div
  className="input-wrapper"
>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-01 MuiFormControl-fullWidth-04"
  >
    <label
      className="MuiFormLabel-root-010 MuiInputLabel-root-05 MuiInputLabel-formControl-06 MuiInputLabel-animated-09"
      data-shrink={false}
      htmlFor={undefined}
    >
      Test label
    </label>
    <div
      className="MuiInput-root-017 MuiInput-fullWidth-024 MuiInput-formControl-018 MuiInput-underline-021"
    >
      <input
        aria-invalid={false}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-025"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={false}
        rows={undefined}
        type="text"
        value={undefined}
      />
    </div>
  </div>
</div>
`;

exports[`<InputField /> should render error status 1`] = `
<div
  className="input-wrapper"
>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-01 MuiFormControl-fullWidth-04"
  >
    <label
      className="MuiFormLabel-root-010 MuiFormLabel-error-013 MuiInputLabel-root-05 MuiInputLabel-formControl-06 MuiInputLabel-animated-09"
      data-shrink={false}
      htmlFor={undefined}
    >
      Test label
    </label>
    <div
      className="MuiInput-root-017 MuiInput-error-022 MuiInput-fullWidth-024 MuiInput-formControl-018 MuiInput-underline-021"
    >
      <input
        aria-invalid={true}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-025"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={false}
        rows={undefined}
        type="text"
        value={undefined}
      />
    </div>
  </div>
  <p
    className="MuiFormHelperText-root-030 MuiFormHelperText-error-031"
  />
</div>
`;
